src/Seq.Input.HealthCheck/HttpHealthCheck.cs

```diff
@@ -124,7 +125,74 @@ public async Task<HealthCheckResult> CheckNow(CancellationToken cancel)
+            if (_bypassHttpCaching)
+                request.Headers.CacheControl = new CacheControlHeaderValue {NoStore = true};
+        }
+
+        private async Task<HttpResponseMessage> MakeAndFollowRequest(CancellationToken cancel, string requestUri, string correlationId,
+            Stack<Redirect> previousRedirects)
+        {
+            var request = new HttpRequestMessage(HttpMethod.Get, requestUri);
+            // include initial headers on every redirect
+            // todo: Should any response headers included in future requests?
+            AddHeadersToRequest(request, correlationId);
+
+            var response = await _httpClient.SendAsync(request, cancel);
+            var statusCode = (int) response.StatusCode;
+            var exceededRedirectCount = previousRedirects.Count > MaxRedirectCount;
+            var responseStatusCodeWas300 = statusCode is >= 300 and <= 399;
+
+            if (_shouldFollowRedirects && responseStatusCodeWas300 && !exceededRedirectCount)
+            {
+                var locationHeader = response.Headers.Location;
+                // https://httpwg.org/specs/rfc9110.html#status.3xx
+                // Redirects that indicate this resource might be available at a different URI:
+                // 301 (Moved Permanently),
+                // 302 (Found),
+                // 303 (See Other),
+                // 307 (Temporary Redirect),
+                // 308 (Permanent Redirect)
+                if (locationHeader is not null)
+                {
+                    // todo: do we want to log the full stack of redirects that occured anywhere?
+                    previousRedirects.Push(new Redirect(request.RequestUri, statusCode, locationHeader));
+                    // todo: Should we rely on the server to include the resource as well as query parameters...
+                    // or should we analyze and ensure that they're included?
+                    var newUri = locationHeader.ToString();
+                    return await MakeAndFollowRequest(cancel, newUri, correlationId, previousRedirects);
+                }
+
+                // todo what to do if there's not a `location` header? as in 300 and 304?
+                // https://httpwg.org/specs/rfc9110.html#status.3xx
+                // Redirection that offers a choice among matching resources capable of representing this resource,
+                // as in the 300 (Multiple Choices) status code.
+                // as in the 304 (Not Modified) status code.
+                else
+                {
+                    // todo: Should this be a warning or an exception?
+                    // todo: Should we destructure the Request & Response?
+                    _log.Warning(
+                        "Http Redirect Status Code {StatusCode} Received with No Location Header {Header} from response {@Response} during request {@Request}",
+                        statusCode,
+                        locationHeader, response, request);
```

Comment:
I think simply failing the check here, by returning `reponse` and not logging anything, would be sufficient. It's (hopefully) an exceedingly rare case.